icu4c/source/common/ulocale.cpp

```diff
@@ -0,0 +1,107 @@
+// © 2023 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+//
+#include "unicode/errorcode.h"
+#include "unicode/stringpiece.h"
+#include "unicode/utypes.h"
+#include "unicode/ustring.h"
+#include "unicode/ulocale.h"
+#include "unicode/locid.h"
+
+#include "cmemory.h"
+
+U_NAMESPACE_USE
+#define EXTERNAL(i) ((ULocale*)(i))
+#define CONST_INTERNAL(e) ((const icu::Locale*)(e))
+#define INTERNAL(e) ((icu::Locale*)(e))
+
+ULocale*
+ulocale_openForLocaleID(const char* localeID, int32_t length) {
+    if (length < 0 || localeID[length] == '\0') {
+        return EXTERNAL(icu::Locale::createFromName(localeID).clone());
```

Comment:
Why do you need the `clone()` here?  And doesn't it mean the result of `createFromName()` leaks?  Or are the locale objects cached or something?

Reply:
The signature of createFromName is

static Locale icu::Locale::createFromName(const char * name) 
It return an object, not an return a pointer nor a reference to an object so I have to use clone() to return a poitner.

Reply:
That's what I was afraid of.  In that case, no objections from me.